pkg/hubble/exporter/api.go

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package exporter
+
+import (
+	"context"
+	"time"
+
+	flowpb "github.com/cilium/cilium/api/v1/flow"
+	v1 "github.com/cilium/cilium/pkg/hubble/api/v1"
+)
+
+type FlowLogExporter interface {
+	OnDecodedEvent(context context.Context, event *v1.Event) (bool, error)
+	Stop() error
+}
+
+type flowLogConfig struct {
+	Name               string               `yaml:"name"`
+	FilePath           string               `yaml:"filePath"`
+	FieldMask          []string             `yaml:"fieldMask"`
+	IncludeFiltersJSON []string             `yaml:"includeFilters"`
+	IncludeFilters     []*flowpb.FlowFilter ``
```

Comment:
I cannot map YAML directly as `flowpb.Flowfilter` as these structs don't have `yaml` annotations and deserialization is not working. That's why I currently have json inside yaml config file.

Reply:
I know gogoprotobuf used to support this and upstream protobuf doesn't, but gogoprotobuf is deprecated. Perhaps we can use something like https://github.com/srikrsna/protoc-gen-gotag?

Reply:
`flowpb.Flowfilter` is a proto generated struct, they currently only have proto and json tags. There may be a protoc plugin that adds yaml tags as well, I have never looked into it

Reply:
https://github.com/cilium/cilium/pull/28943
https://github.com/cilium/cilium/pull/28944

Reply:
I replaced yaml parser with `sigs.k8s.io/yaml` which uses json tags.